v21.2/performance-best-practices-overview.md

```diff
@@ -393,35 +368,14 @@ To avoid contention, multiple strategies can be applied:
-
-- Increase
-  [normalization](https://en.wikipedia.org/wiki/Database_normalization)
-  of the data to place parts of the same records that are modified by
-  different transactions in different tables. Note however that this
-  is a double-edged sword, because denormalization can also increase
-  performance by creating multiple copies of often-referenced data in
-  separate ranges.
-
-- If the application strictly requires operating on very few different
-  index key values, consider using [`ALTER ... SPLIT
-  AT`](split-at.html) so that each index key value can be served by
-  a separate group of nodes in the cluster.
-
-It is always best to avoid contention as much as possible via the
-design of the schema and application. However, sometimes contention is
-unavoidable. To maximize performance in the presence of contention,
-you'll need to maximize the performance of a single range.
-To achieve this, multiple strategies can be applied:
-
-- Minimize the network distance between the replicas of a range,
-  possibly using zone configs and partitioning.
+- When replacing values in a row, use [`UPSERT`](upsert.html) and specify values for all columns in the inserted rows. This will usually have the best performance under contention, compared to combinations of [`SELECT`](select-clause.html), [`INSERT`](insert.html), and [`UPDATE`](update.html).
+
+- Increase [normalization](https://en.wikipedia.org/wiki/Database_normalization) of the data to place parts of the same records that are modified by different transactions in different tables. Increasing normalization has drawbacks as well as benefits, because denormalization can also increase performance by creating multiple copies of often-referenced data in separate ranges.
+
+- If the application strictly requires operating on very few different index key values, consider using [`ALTER ... SPLIT AT`](split-at.html) so that each index key value can be served by a separate group of nodes in the cluster.
+
+It is always best to avoid contention as much as possible via the design of the schema and application. However, sometimes contention is unavoidable. To maximize performance in the presence of contention, you'll need to maximize the performance of a single range. To achieve this, multiple strategies can be applied:
+
+- Minimize the network distance between the replicas of a range, possibly using zone configs and partitioning.
 - Use the fastest storage devices available.
-- If the contending transactions operate on different keys within the
-  same range, add more CPU power (more cores) per node. Note however
-  that this is less likely to provide an improvement if the
-  transactions all operate on the same key.
+- If the contending transactions operate on different keys within the same range, add more CPU power (more cores) per node. Note however that this is less likely to provide an improvement if the transactions all operate on the same key.
```

Comment:
```suggestion
- If the contending transactions operate on different keys within the same range, add more CPU power (more cores) per node. However, if the transactions all operate on the same key, this may not provide an improvement .
```